algoliasearch-core/src/main/java/com/algolia/search/models/settings/IndexSettings.java

```diff
@@ -670,6 +671,20 @@ public IndexSettings setCustomSettings(Map<String, Object> customSettings) {
     return this;
   }
 
+  public Boolean getEnablePersonalization() {
+    return enablePersonalization != null && enablePersonalization;
+  }
+
+  public IndexSettings setEnablePersonalization(Boolean enablePersonalization) {
+    this.enablePersonalization = enablePersonalization;
+    return this;
+  }
+
+  @JsonProperty("enablePersonalization")
```

Comment:
You don't need `@JsonProperty` here because Jackson will (by default) serialize the property with its name, here defined line 32.

Reply:
The problem is that the property will always get serialized either because it has been set or the default value, the following test, for example, will start to fail:: https://github.com/algolia/algoliasearch-client-java-2/blob/92e22dadba7e1b2e589f20688003a2e34ffd54e0/algoliasearch-core/src/test/java/com/algolia/search/JacksonParserTest.java#L29

please check https://github.com/algolia/algoliasearch-client-java-2/pull/700#discussion_r436768692 and tell me what you think?